src/dattobd.c

```diff
@@ -1785,6 +1785,157 @@ static void tp_put(struct tracing_params *tp){
 
 /****************************BIO HELPER FUNCTIONS*****************************/
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 3, 0))
```

Comment:
Due to the many different distributions of linux, each with their own feature backports and tweaks we have discovered that it is necessary to rely on feature detection rather than version numbers to decide whether to include code. See genconfig.sh, which runs a series of dummy tests to produce includes.h, which defines a set of macros that represent the features of the kernel.